setup.py

```diff
@@ -23,13 +23,14 @@
     "qiskit-terra>=0.13.0",
     "retworkx>=0.8.0",
     "scipy>=0.19,!=0.19.1",
-    "setuptools>=40.1.0",
+    "setuptools>=40.1.0"
+    "scikit-learn>=0.24.2"
```

Comment:
scikit-learn is already an optional dependency. I don't think this feature along with the iq discriminator are really widely used enough to warrant making a hard requirement for all users. I think we should leave it as an optional dependency and maybe add a new new name for the optional extra instead of just `iq`.

Reply:
I used scikit-learn for cluster decoder as a means for using clustering algorithms and also adding more clustering algorithms in the future but I can make it optional.

Reply:
Made it optional. And removed cluster decoder testing to save it from possible failure.